### ISSUE TITLE
pip install fails on Alpine linux

### ISSUE THREAD

**Issue by sflanker** [wseaton/sqloxide#252]
Running `pip install sqloxide` fails on Alpine linux (using `python:3.8-alpine` docker image):

```
# pip install sqloxide
Collecting sqloxide
  Downloading sqloxide-0.1.36.tar.gz (7.8 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: sqloxide
  Building wheel for sqloxide (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for sqloxide (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      /usr/local/bin/python: can't open file 'build.py': [Errno 2] No such file or directory
      Traceback (most recent call last):
        File "/usr/local/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-b_woof1b/overlay/lib/python3.8/site-packages/poetry/core/masonry/api.py", line 57, in build_wheel
          return WheelBuilder.make_in(
        File "/tmp/pip-build-env-b_woof1b/overlay/lib/python3.8/site-packages/poetry/core/masonry/builders/wheel.py", line 87, in make_in
          wb.build(target_dir=directory)
        File "/tmp/pip-build-env-b_woof1b/overlay/lib/python3.8/site-packages/poetry/core/masonry/builders/wheel.py", line 122, in build
          self._build(zip_file)
        File "/tmp/pip-build-env-b_woof1b/overlay/lib/python3.8/site-packages/poetry/core/masonry/builders/wheel.py", line 171, in _build
          self._run_build_script(self._package.build_script)
        File "/tmp/pip-build-env-b_woof1b/overlay/lib/python3.8/site-packages/poetry/core/masonry/builders/wheel.py", line 231, in _run_build_script
          subprocess.check_call([self.executable.as_posix(), build_script])
        File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['/usr/local/bin/python', 'build.py']' returned non-zero exit status 2.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for sqloxide
Failed to build sqloxide
ERROR: Could not build wheels for sqloxide, which is required to install pyproject.toml-based projects
```

**Comment by sflanker** [comment]
I've tried with `setuptools-57.5.0` and with `setuptools-68.1.2`
Pip version: 23.2.1

**Comment by wseaton** [comment]
@sflanker can you provide the rest of your Dockerfile? 

I can reproduce the same issue with the following:
```Dockerfile
FROM python:3.8-alpine

RUN apk add curl && apk add gcc
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y

ENV PATH $HOME/.cargo/bin:$PATH
RUN pip install --upgrade pip && \
    pip install --upgrade setuptools && \
    pip install --upgrade setuptools-rust && \
    pip install sqloxide

```

Alpine requires building from source because it's a musl target. Will need to investigate if it's possible to support this target from either `setuptools-rust` or `maturin`.  Not quite sure why the actual build from source is broken on alpine though.

**Comment by wseaton** [comment]
Ah okay, looks like the build script is actually missing from the source distribution: https://files.pythonhosted.org/packages/51/c3/dda014064889c6065d597c4ce8dae30f87992bd79f9771349ac7bdd890d4/sqloxide-0.1.36.tar.gz. Should be a relatively simple fix.

**Comment by wseaton** [comment]
@sflanker this should hopefully be fixed in the `v0.1.37` release that I just triggered (should land in a few minutes). Please let me know if it isn't, but as a workaround you should be able to build from git, eg. `pip install git+{repo_url}` as that will always include the build.py with the source.
